cranelift/codegen/src/ir/immediates.rs

```diff
@@ -1189,7 +1193,8 @@ impl Ieee64 {
     /// Returns the nearest integer to `self`. Rounds half-way cases to the number
     /// with an even least significant digit.
     pub fn round_ties_even(self) -> Self {
-        Self::with_float(self.as_f64().round_ties_even())
+        // TODO: find a suitable alternative to std round_ties_even() in no_std
+        Self::with_float(self.as_f64().round())
```

Comment:
I looked at the [source for `round_ties_even`](https://doc.rust-lang.org/src/std/f64.rs.html#127-129) and they use `rintf`. It looks like this is [due to portability between platforms](https://github.com/rust-lang/rust/pull/95317#issuecomment-1306215792), but it looks like it gives the same results if we don't change rounding modes. `libm` has support for `rint{f,}` so we can probably use that.

Reply:
We'll want to be careful though such that where `std` is enabled we want the same implementation cranelift has today which is using presumably optimal routines in the standard library. Only when that feature is disabled should it fall back to the possibly-slower routines based in `libm`